sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3624,7 +3626,8 @@ object GpuOverrides extends Logging {
     part[RangePartitioning](
       "Range partitioning",
       PartChecks(RepeatingParamCheck("order_key",
-        (pluginSupportedOrderableSig + TypeSig.DECIMAL_128 + TypeSig.STRUCT).nested(),
+        (pluginSupportedOrderableSig + TypeSig.DECIMAL_128 + TypeSig.STRUCT).nested() +
+         TypeSig.ARRAY.nested(_gpuCommonTypes + TypeSig.DECIMAL_128),
```

Comment:
Sadly this is not correct. The TypeSig checks don't have a way to distinguish between child types. So by adding ARRAY to the top level sigs you also are allowing `ARRAY[STRUCT[....]]`  The only way to fix this is to add a PsTag to ARRAY to update the docs to say that STRUCT is not supported as a child type for ARRAYs, and then add code to check for this in the `tagExprForGpu` method.

And with that could you add some tests to verify that we fall back to the CPU for sorts of array[struct[Something]]

Reply:
Thank you @revans2 ! Changes updated.